src/Core/Services/UsernameGenerationService.cs

```diff
@@ -0,0 +1,250 @@
+            if (options.RandomWordUsernameCapitalize == null)
+            {
+                options.RandomWordUsernameCapitalize = false;
+            }
+            if (options.RandomWordUsernameIncludeNumber == null)
+            {
+                options.RandomWordUsernameIncludeNumber = false;
+            }
+            var listLength = EEFLongWordList.Instance.List.Count - 1;
+            var wordIndex = await _cryptoService.RandomNumberAsync(0, listLength);
+            if (options.RandomWordUsernameCapitalize.GetValueOrDefault())
+            {
+                randomWord = Capitalize(EEFLongWordList.Instance.List[wordIndex]);
+            }
+            else
+            {
+                randomWord = EEFLongWordList.Instance.List[wordIndex];
+            }
+
+            if (options.RandomWordUsernameIncludeNumber.GetValueOrDefault())
+            {
+                randomWord = await AppendRandomNumberToRandomWordAsync(randomWord);
+            }
+
+            return randomWord;
+        }
+
+        private async Task<string> GeneratePlusAddressedEmailAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+            string generatedString = string.Empty;
+            var adressedEmail = options.PlusAddressedEmail;
+
+            if (string.IsNullOrWhiteSpace(adressedEmail) || adressedEmail.Length < 3)
+            {
+                return DefaultGenerated;
+            }
+
+            var atIndex = adressedEmail.IndexOf("@");
+            if (atIndex < 1 || atIndex >= adressedEmail.Length - 1)
+            {
+                return adressedEmail;
+            }
+
+            var emailBeginning = adressedEmail.Substring(0, atIndex);
+            var emailEnding = string.Empty;
+
+            emailEnding = adressedEmail.Substring(atIndex + 1, adressedEmail.Length - (atIndex + 1));
+
+            if (options.PlusAddressedEmailType == UsernameEmailType.Random)
+            {
+                generatedString = await RandomStringAsync(8);
+            }
+            else
+            {
+                generatedString = options.EmailWebsite;
+            }
+
+            return emailBeginning + "+" + generatedString + "@" + emailEnding;
+        }
+
+        private async Task<string> GenerateCatchAllAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+            var catchAllEmailDomain = options.CatchAllEmailDomain;
+            string generatedString = string.Empty;
+
+            if (string.IsNullOrWhiteSpace(catchAllEmailDomain))
+            {
+                return DefaultGenerated;
+            }
+
+            if (options.CatchAllEmailType == UsernameEmailType.Random)
+            {
+                generatedString = await RandomStringAsync(8);
+            }
+            else
+            {
+                generatedString = options.EmailWebsite;
+            }
+
+            return generatedString + "@" + catchAllEmailDomain;
```

Comment:
return directly in the if, you can define the const at the top of the class if you want:
```c#
const CATCH_ALL_EMAIL_DOMAIN_FORMAT = "{0}@{1}";

if (options.CatchAllEmailType == UsernameEmailType.Random)
{
    var randomString = await RandomStringAsync(8);
    return string.Format(CATCH_ALL_EMAIL_DOMAIN_FORMAT, randomString, catchAllEmailDomain);
}

return string.Format(CATCH_ALL_EMAIL_DOMAIN_FORMAT, options.EmailWebsite, catchAllEmailDomain);
```